EIPS/eip-7742.md

```diff
@@ -44,37 +44,68 @@ and deployment of changes to the blob count easier.
+
+Note: CLs must also ensure that the target is always equal to or more than half the max otherwise we will need to also track the max for normalizing our gas calculations (otherwise full blocks will make the price jump by a factor of more than 1.125)
 
 ## Specification
 
-### Block structure and validity
+### Execution Block structure and validity
 
 Upon activation of this EIP, execution clients **MUST** extend the header schema with an
-additional 64-bit field: the `target_blob_count`. This value is set to the current target blob count. The Engine API
-is modified along with this EIP to provide the `target_blob_count` with each payload and implementations can use this
-value to correctly set the block header field.
+additional 64-bit field: the `target_blob_count`. This values is set to the one provided by the CL with the modified engine api fcU's payload attributes.
+
+Validity of these values is guaranteed from the consensus layer, much like how withdrawals are handled. Hence when verifying a block, execution clients **MUST** ensure the target blob count in the block header matches the one provided by the consensus client.
+
+For a genesis block with no existing parent, the value should be set according to the agreed specification for the target blob count given by that genesis block's protocol rule set.
+
+`excess_blob_gas` in the block now holds _normalized_ excess blob gas accumulating `((blob_gas_used-target_blob_gas)/target_blob_gas) * EXCESS_BLOB_GAS_NORMALIZATION_FACTOR`. We choose `EXCESS_BLOB_GAS_NORMALIZATION_FACTOR` to `100E6` and this makes our choice of `NORMALIZED_BLOB_BASE_FEE_UPDATE_FRACTION` as `3338477 * 100E6 // 393216 = 849018605`.
+
+| Constant | Value |
+| - | - |
+| `EXCESS_BLOB_GAS_NORMALIZATION_FACTOR` | `100E6` |
+| `NORMALIZED_BLOB_BASE_FEE_UPDATE_FRACTION` | `849018605` |
+| `OLD_TARGET_BLOB_GAS_PER_BLOCK` | `393216` |
+
+At the fork block (where parent.timestamp < FORK_TIMESTAMP), we convert parent's accumulated `excess_blob_gas` to the normalized `excess_blob_gas` to be used in this block's `excess_blob_gas` accumulator.
+
+```python
+def calc_excess_blob_gas(block: Header) -> int:
+  if(block.parent.timestamp < FORK_TIMESTAMP)
+    normalizedParentExcessBlobGas = block.parent.excess_blob_gas * EXCESS_BLOB_GAS_NORMALIZATION_FACTOR // OLD_TARGET_BLOB_GAS_PER_BLOCK
+    blockTargetBlobGas = TARGET_BLOB_GAS_PER_BLOCK
+  else
+    normalizedParentExcessBlobGas = block.parent.excess_blob_gas
+    blockTargetBlobGas = block.target_blob_count * GAS_PER_BLOB
 
-Validity of the `target_blob_count` is guaranteed from the consensus layer, much like how withdrawals are handled.
+  return (normalizedParentExcessBlobGas + (block.blob_gas_used - blockTargetBlobGas) * EXCESS_BLOB_GAS_NORMALIZATION_FACTOR // blockTargetBlobGas)
```

Comment:
In 4844, this function was `def calc_excess_blob_gas(parent: Header)` - this change now mixes data from the current block with the parent. Wondering what the rationale is for using `block.blob_gas_used` and `block.target_blob_count` is instead of the `parent` values?

Reply:
If it's possible use parent for all three values, that would simplify the changes, in Besu at least.

This might also impact eth_feeHistory since we'd need data external to the block in order to calculate the next fee

Reply:
ahh yes can shift to that arrangement

Reply:
updated @siladu  and thanks! 
can you check it it all looks good now?